src/main/java/org/talend/component/api/configuration/ui/widget/NestedProperties.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) 2006-2017 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.component.api.configuration.ui.widget;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.talend.component.api.configuration.ui.meta.Ui;
+import org.talend.component.api.meta.Documentation;
+
+@Ui
+@Documentation("Mark a field as being represented by a nested properties widget - list of properties that can be "
+        + "added or removed. Can be represented as a horizontal table widget in studio.")
+@Retention(RUNTIME)
+@Target({ FIELD, PARAMETER })
+public @interface NestedProperties {
```

Comment:
Normally we don't need it since it will already be identified by the fact the field is a List<Something>